File: server/models/blog.js
const mongoose = require("mongoose");

const blogSchema = mongoose.Schema({
    title: {
        required: true,
        type: String,
    },
    content: {
        required: true,
        type: String,
    },
    ImageUrl: {
        type: String
    },
    topics: {
        require: true,
        type: Array,
    }
});

const Blog = mongoose.model("Blog", blogSchema);

module.exports = Blog;

File: blogapp/lib/features/blog/domain/domain/entities/blog.dart
class Blog {
  final String id;
  final String title;
  final List<String> topic;
  final String content;
  final String imageUrl;
  final String posterId;
  final DateTime updatedAt;

  Blog({
    required this.id,
    required this.title,
    required this.topic,
    required this.content,
    required this.imageUrl,
    required this.posterId,
    required this.updatedAt,
  });
}
